docker/configure_workers_and_start.py

```diff
@@ -469,32 +586,218 @@ def generate_worker_files(
 
     # A counter of worker_type -> int. Used for determining the name for a given
     # worker type when generating its config file, as each worker's name is just
-    # worker_type + instance #
-    worker_type_counter: Dict[str, int] = {}
+    # worker_type(s) + instance #
```

Comment:
I know it was already fairly poor before you came to it, but `generate_worker_files` is now monstruously long.

It might be better if the logic for parsing and validation for the worker type string was pulled out into a function that accepts the worker description string and returns

`Dict[str, Set[str]]` — mapping from worker name to set of worker roles

?

I'd try and keep it quite concise/punchy because it's hard to read such long chains of processing.

Reply:
Hopefully it's closer to what you asked for. I did my best.